local-interchain/interchain/handlers/container_log_stream.go

```diff
@@ -0,0 +1,153 @@
+package handlers
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+	"unicode"
+
+	dockertypes "github.com/docker/docker/api/types"
+	dockerclient "github.com/docker/docker/client"
+	"github.com/strangelove-ventures/interchaintest/v8/chain/cosmos"
+	"go.uber.org/zap"
+)
+
+var removeColorRegex = regexp.MustCompile("\x1b\\[[0-9;]*m")
+
+type ContainerStream struct {
+	ctx      context.Context
+	logger   *zap.Logger
+	cli      *dockerclient.Client
+	authKey  string
+	testName string
+
+	nameToID map[string]string
+}
+
+func NewContainerSteam(ctx context.Context, logger *zap.Logger, cli *dockerclient.Client, authKey, testName string, vals map[string][]*cosmos.ChainNode) *ContainerStream {
+	nameToID := make(map[string]string)
+	for _, nodes := range vals {
+		for _, node := range nodes {
+			nameToID[node.Name()] = node.ContainerID()
+		}
+	}
+
+	return &ContainerStream{
+		ctx:      ctx,
+		authKey:  authKey,
+		cli:      cli,
+		logger:   logger,
+		testName: testName,
+		nameToID: nameToID,
+	}
+}
+
+func (cs *ContainerStream) StreamContainer(w http.ResponseWriter, r *http.Request) {
+	// ensure ?auth_key=<authKey> is provided
+	if cs.authKey != "" && r.URL.Query().Get("auth_key") != cs.authKey {
+		http.Error(w, "Unauthorized, incorrect or no ?auth_key= provided", http.StatusUnauthorized)
+		return
+	}
+
+	containerID := r.URL.Query().Get("id")
+	if containerID == "" {
+		output := "No container ID provided. Available containers:\n"
+		for name, id := range cs.nameToID {
+			output += fmt.Sprintf("- %s: %s\n", name, id)
+		}
+
+		fmt.Fprint(w, output)
+		fmt.Fprint(w, "Provide a container ID with ?id=<containerID>")
+		return
+	}
+
+	// if container id is in the cs.nameToID map, use the mapped container ID
+	if id, ok := cs.nameToID[containerID]; ok {
+		containerID = id
+	} else {
+		fmt.Fprintf(w, "Container ID %s not found\n", containerID)
+		return
+	}
+
+	// http://127.0.0.1:8080/container_logs?id=<ID>&colored=true
+	isColored := strings.HasPrefix(strings.ToLower(r.URL.Query().Get("colored")), "t")
+	tailLines := tailLinesParam(r.URL.Query().Get("lines"))
+
+	rr, err := cs.cli.ContainerLogs(cs.ctx, containerID, dockertypes.ContainerLogsOptions{
+		ShowStdout: true,
+		ShowStderr: true,
+		Follow:     true,
+		Details:    true,
+		Tail:       strconv.FormatUint(tailLines, 10),
+	})
+	if err != nil {
+		http.Error(w, "Unable to get container logs", http.StatusInternalServerError)
+		return
+	}
+	defer rr.Close()
+
+	// Set headers to keep the connection open for SSE (Server-Sent Events)
+	w.Header().Set("Content-Type", "text/event-stream")
+	w.Header().Set("Cache-Control", "no-cache")
+	w.Header().Set("Connection", "keep-alive")
+
+	// Flush ensures data is sent to the client immediately
+	flusher, ok := w.(http.Flusher)
+	if !ok {
+		http.Error(w, "Streaming unsupported", http.StatusInternalServerError)
+		return
+	}
+
+	for {
+		buf := make([]byte, 8*1024)
```

Comment:
What was the reasoning for instantiating the slice with this specific value? Does changing this value break anything? If so a comment regarding the reason for this specific size could be helpful.

Reply:
1024 is the standard but due to the size of some JSON blobs sent through it was exceeding that for Galen and he was unable to properly parse (as it came in multiple chunks). 

It was exceeding the size for some relayer JSON blobs for Galen. He was unable to parse it since it came in multiple chunks. I wanted to ensure we have enough of a buffer and 8192 felt like a good spot (2*1024 was still too small, I am sure 4* would have been fine but increased just to be sure since the data sent it relative small and controlled anyways)

Reply:
Not necessary but this is the type of decision that can be helpful to document via a comment. You never know when someone is going to come across and naively change values without understanding what the implications are.